src/Education/CrossReferenceRuleHelper.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Education
+{
+    /// <summary>
+    /// This is a helper class to encode a SonarCompositeRuleId into a valid URI and decode it vice versa.
+    /// </summary>
+    internal static class CrossReferenceRuleHelper
+    {
+        private const string prefix = "sonarlintrulecrossref";
```

Comment:
When creating a URI all capitalization gets removed, to make it less confusing I removed it here too simply so that when we test and when we decode we won't have to think about capitalization.

Reply:
What happened to `ahhhhh` ?! ;-)